File: slide/src/main/java/cangwang/com/slide/SlideFragment.java
package cangwang.com.slide;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

/**侧边弹框
 * Created by cangwang on 2018/2/10.
 */

public class SlideFragment extends Fragment{
    public static final String TAG = "SlideFragment";
    private View view;

    @Override
    public Animation onCreateAnimation(int transit, boolean enter, int nextAnim) {
        if (enter) {
            return AnimationUtils.loadAnimation(getActivity(), R.anim.slide_in_from_right);
        } else {
            return AnimationUtils.loadAnimation(getActivity(), R.anim.slide_out_from_left);
        }
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.slide_fragment_layout,container,false);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

            }
        });
        return view;
    }

}

File: slide/src/main/java/cangwang/com/slide/SlideFragment.kt
package cangwang.com.slide

import android.os.Bundle

import android.view.animation.Animation
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import androidx.fragment.app.Fragment

/**侧边弹框
 * Created by cangwang on 2018/2/10.
 */
class SlideFragment : Fragment() {
    private var slideView: View? = null
    override fun onCreateAnimation(transit: Int, enter: Boolean, nextAnim: Int): Animation? {
        return if (enter) {
            AnimationUtils.loadAnimation(activity, R.anim.slide_in_from_right)
        } else {
            AnimationUtils.loadAnimation(activity, R.anim.slide_out_from_left)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        slideView = inflater.inflate(R.layout.slide_fragment_layout, container, false)
        slideView?.setOnClickListener(View.OnClickListener { })
        return view
    }

    companion object {
        const val TAG = "SlideFragment"
    }
}